docs/tutorials/qsimcirq_gcp.md

```diff
@@ -0,0 +1,357 @@
+*   Choose the machine family / series: N2
+    *   Quantum simulation can require a high level of resources, so we are selecting powerful processors, but [many choices are available.](https://cloud.google.com/blog/products/compute/choose-the-right-google-compute-engine-machine-type-for-you)
+
+![alt_text](../images/image13.png )
+
+*   Choose the [Machine Type](https://cloud.google.com/blog/products/compute/choose-the-right-google-compute-engine-machine-type-for-you): n2-standard-16
+    *   16 CPUs
+    *   64GB memory
+*   Choose the Boot Disk image:[ Container-Optimized OS](https://cloud.google.com/container-optimized-os/docs/concepts/features-and-benefits)
+    *   Leave the remaining as defaults.
+
+![alt_text](../images/image10.png )
+
+Finally, enable HTTP access
+*   Click "Create"
+
+![alt_text](../images/image8.png )
+
+## Preparing your local system to connect to Colab
+[Colaboratory](https://colab.sandbox.google.com/notebooks/intro.ipynb) or "Colab" for short, allows you to write and execute Python in your browser, with
+
+* Zero configuration required
+* Free access to GPUs
+* Easy sharing
+
+We will focus on how to run simulations GCE VM your created above using Colab as the UI / frontend.
+
+### Install Gcloud on your local machine
+In order to connnect your VM to Colab, you need to forward the default Jupyter port (8888) to your localhost. Port forwarding to a GCP VM requires that you have th Google Cloud Platform SDK installed on your local computer. This will give you access to the `gcloud` command.
+
+Intallation instructions a be found [here](https://cloud.google.com/sdk/docs/install).
+
+After installing, initialize the Google Cloud environment.
+
+```bash
+gcloud init
+```
+Respond to the prompts with your project ID and the zone where you created your VM. When completed, view the glcoud configuration:
+
+```bash
+gcloud config list
+```
+Output should look something like: 
+
+```less
+region = us-east4
+zone = us-east4-c
+[core]
+account = wellhello@gargle.com
+disable_usage_reporting = False
+project = quantum-22222
+```
+
+## Connect via port forward to your VM
+You are now ready co connect to your VM via a port forward gcloud command. Detailed instructions can be found [here](https://research.google.com/colaboratory/local-runtimes.html), but the following instructions should be sufficient to succeed. Connect up with SSH with port 8888 forwarded to your local machine:
+
+```bash
+gcloud compute ssh [YOUR_INSTANCE_NAME] -- -L 8888:localhost:8888
+```
+You should now see the command line prompt from your VM:
+```bash
+wellhello@qsim-1 ~ $ 
+```
+### Run the Jupyter / qsim container on your VM 
+At the command prompt you can now start a Docker container with all the required code to run simulations. Start the container:
+
+```bash
+docker run -v `pwd`:/homedir -p 8888:8888 gcr.io/quantum-builds/github.com/quantumlib/jupyter_qsim:latest
```

Comment:
`docker {...}` returns errors if you're not running as root. We should highlight this possibility and remind users to try `sudo docker {...}` if they have issues.

Reply:
That's unexpected. Is there an error you can share?

Reply:
Usually something of this form:
```
ERROR: Got permission denied while trying to connect to the Docker daemon socket at \
unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.40/info: \
dial unix /var/run/docker.sock: connect: permission denied
```
As far as I know, this is [expected behavior](https://docs.docker.com/engine/install/linux-postinstall/#:~:text=The%20Docker%20daemon%20always%20runs,members%20of%20the%20docker%20group.) for non-root users of Docker, but it wouldn't hurt to have a reminder since the error message is pretty opaque.

Reply:
When you run `id` do you find the `docker` group?
```
id
uid=20141(jrossthomson) gid=20142(jrossthomson) groups=20142(jrossthomson),4(adm),27(video),412(docker),20141(google-sudoers)
```

Reply:
> When you run `id` do you find the `docker` group?

No. (I'm aware that there are ways to set up Docker that eliminate the need for `sudo`, but I haven't done this in my environment and I don't think we should assume that all users have, either.)

Reply:
I don't think we want to cover too much docker install here. Probably fine to present the straight-forward usage and add a sentence/aside to look at [this page](https://docs.docker.com/engine/install/linux-postinstall/) for instructions to run Docker as a non-root user.